opslevel/resource_opslevel_team.go

```diff
@@ -28,49 +28,59 @@ func NewTeamResource() resource.Resource {
 	return &TeamResource{}
 }
 
-// TeamResourceModel describes the Team managed resource.
-type TeamResourceModel struct {
-	Aliases          types.List   `tfsdk:"aliases"`
-	Id               types.String `tfsdk:"id"`
-	LastUpdated      types.String `tfsdk:"last_updated"`
-	Member           []TeamMember `tfsdk:"member"`
-	Name             types.String `tfsdk:"name"`
-	Parent           types.String `tfsdk:"parent"`
-	Responsibilities types.String `tfsdk:"responsibilities"`
+// teamResourceModel describes the Team managed resource.
+type teamResourceModel struct {
+	Aliases          types.List        `tfsdk:"aliases"`
+	Id               types.String      `tfsdk:"id"`
+	LastUpdated      types.String      `tfsdk:"last_updated"`
+	Member           []teamMemberModel `tfsdk:"member"`
+	Name             types.String      `tfsdk:"name"`
+	Parent           types.String      `tfsdk:"parent"`
+	Responsibilities types.String      `tfsdk:"responsibilities"`
 }
 
-type TeamMember struct {
-	Email types.String `tfsdk:"email"`
-	Role  types.String `tfsdk:"role"`
-}
-
-func convertTeamMember(teamMember opslevel.TeamMembership) TeamMember {
-	return TeamMember{
-		Email: RequiredStringValue(teamMember.User.Email),
-		Role:  RequiredStringValue(teamMember.Role),
-	}
-}
-
-func NewTeamResourceModel(ctx context.Context, team opslevel.Team) (TeamResourceModel, diag.Diagnostics) {
+func newTeamResourceModel(ctx context.Context, team opslevel.Team, parentIdentifier string) (teamResourceModel, diag.Diagnostics) {
 	aliases, diags := OptionalStringListValue(ctx, team.ManagedAliases)
 	if diags != nil && diags.HasError() {
-		return TeamResourceModel{}, diags
+		return teamResourceModel{}, diags
 	}
-	teamMembers := make([]TeamMember, 0)
+	teamMembers := make([]teamMemberModel, 0)
 	if team.Memberships != nil {
 		for _, mem := range team.Memberships.Nodes {
-			teamMembers = append(teamMembers, convertTeamMember(mem))
+			teamMembers = append(teamMembers, newTeamMemberModel(mem))
 		}
 	}
-	teamResourceModel := TeamResourceModel{
+	model := teamResourceModel{
 		Aliases:          aliases,
 		Id:               ComputedStringValue(string(team.Id)),
 		Member:           teamMembers,
 		Name:             RequiredStringValue(team.Name),
 		Responsibilities: OptionalStringValue(team.Responsibilities),
 	}
+	// set parent team
+	if team.ParentTeam.Id == "" {
+		model.Parent = types.StringNull()
+	} else if opslevel.IsID(parentIdentifier) {
+		model.Parent = types.StringValue(string(team.ParentTeam.Id))
+	} else {
+		// can use non-default aliases
+		model.Parent = types.StringValue(parentIdentifier)
+	}
 
-	return teamResourceModel, diags
+	return model, diags
+}
+
+// removeNonTerraformManagedMembers mutates the input model to exclude team members not managed by terraform
+func removeNonTerraformManagedMembers(ctx context.Context, model *teamResourceModel, cachedModel teamResourceModel) {
```

Comment:
Would there be a way to write this so we can put this function under unit tests?  Then we could write table tests to ensure all the different combinations work as expected?

Reply:
Filed ticket for this - https://github.com/OpsLevel/team-platform/issues/365